torch/testing/_internal/common_methods_invocations.py

```diff
@@ -845,6 +845,88 @@ def sample_inputs_max_min_reduction_no_dim(op_info, device, dtype, requires_grad
                                           requires_grad=requires_grad),))
     return inputs
 
+# Missing to test the nondeterminism of the operation
+# https://github.com/pytorch/pytorch/issues/53352
+def sample_inputs_put(op_info, device, dtype, requires_grad):
+    def make_arg(shape, contiguous=True):
+        return make_tensor(shape, device=device, dtype=dtype,
+                           low=None, high=None,
+                           discontiguous=not contiguous,
+                           requires_grad=requires_grad)
+
+    src = make_arg((S,))
+    src_nonctg = make_arg((S,), contiguous=False)
+    tgt = make_arg((S, S))
+    # non-contiguous tgt
+    tgt_nonctg = tgt.transpose(0, 1)
```

Comment:
why are you making non-contiguous target as a transpose rather than through make_tensor with discontiguous arg?

Reply:
To reduce the memory footprint. That being said, given that the tensors are small, should I just create a new one?

Reply:
I updated the tests to have the much nicer and homogeneous generator pattern. In particular, I generated everything using `make_tensor`.